### PR TITLE
feat: Added API and unit tests to facilitate token mint, token associating, and granting kyc for TokenCreateCustom contract

### PR DESCRIPTION
**Pull request by quiet-node**
**Description**:
This PR adds a set of APIs and unit tests to support facilitating interacting with TokenCreateCustom contract. The methods exposed in this PR are

- mintTokenPublic()
- mintTokenToAddressPublic()
- mintNonFungibleTokenToAddressPublic()
- associateTokenPublic
- associateTokensPublic()
- grantTokenKycPublic()

**Related issue(s)**: #314

Fixes partial #316

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: system-contract-dapp-playground/src/api/hedera/tokenCreateCustom-interactions/index.ts [hunk 1]
```diff
@@ -239,8 +239,236 @@ export const createHederaNonFungibleToken = async (
...
+ *
+ * @return Promise<TokenCreateCustomSmartContractResult>
+ */
+export const mintHederaTokenToAddress = async (
+  baseContract: Contract,
+  tokenType: 'FUNGIBLE' | 'NON_FUNGIBLE',
+  hederaTokenAddress: string,
+  recipientAddress: string,
+  amountToMint: number,
+  metadata: string
```
Comment (Nana-EC) [inline system-contract-dapp-playground/src/api/hedera/tokenCreateCustom-interactions/index.ts:326]: Q: shouldn't this be collection of metadata to support minting more than 1 NFT?

File: system-contract-dapp-playground/__tests__/hedera/hts-interactions/index.test.ts [hunk 2]
```diff
@@ -94,6 +101,36 @@ describe('createHederaFungibleToken test suite', () => {
         hash: txHash,
       }),
     }),
+    mintTokenPublic: jest.fn().mockResolvedValue({
```
Comment (Nana-EC) [inline system-contract-dapp-playground/__tests__/hedera/hts-interactions/index.test.ts:104]: With the `txHash` being the same for all maybe you should add an additional field that confirm the type of transaction.
Maybe similar to the transaction above there should be an event emitted you confirm

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Nice setup and progress.
1 question 1 suggestion

File: system-contract-dapp-playground/__tests__/hedera/hts-interactions/index.test.ts (hunk 2 above)
Comment (quiet-node) [reply system-contract-dapp-playground/__tests__/hedera/hts-interactions/index.test.ts:108]: Agree! I have put in the event filter for the token mint methods. For association and KYC, as the methods emit only ResponseCode events, and the APIs don't utilize these events so I thought I might have leave these out

**Review by Nana-EC — DISMISSED** [review]

File: system-contract-dapp-playground/src/api/hedera/tokenCreateCustom-interactions/index.ts [hunk 3]
```diff
@@ -239,8 +239,258 @@ export const createHederaNonFungibleToken = async (
...
+ *
+ * @param hederaTokenAddress: string
+ *
+ * @param amountToMint: number
+ *
+ * @param metadata: string[]
+ *
+ * @return Promise<TokenCreateCustomSmartContractResult>
+ */
+export const mintHederaToken = async (
```
Comment (Nana-EC) [inline system-contract-dapp-playground/src/api/hedera/tokenCreateCustom-interactions/index.ts:265]: nit: I feel like these should all be in there own files and brought together in index.ts but that can be done later

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**